__init__.py

```diff
@@ -163,12 +163,19 @@ def initialize(self):
 
         # Support query for active alarms from other skills
         self.add_event("private.mycroftai.has_alarm", self.on_has_alarm)
+        self.add_event("skill.alarm.query-active", self.handle_active_alarm_query)
 
     def on_has_alarm(self, message):
         """Reply to requests for alarm on/off status."""
         total = len(self.settings["alarm"])
         self.bus.emit(message.response(data={"active_alarms": total}))
 
+    def handle_active_alarm_query(self, _):
+        """Event handler for the skill.alarm.query-active command."""
+        event_data = {"active_alarms": bool(self.settings["alarm"])}
+        event = Message("skill.alarm.active-queried", data=event_data)
```

Comment:
should this be a `message.response` rather than a new Message?

Reply:
What is the definition of "active alarm"? Any alarm at all (like for next week, or next year)? Or just in the next 24 hrs? Or just until midnight tonight? I don't care what the answer is... just that it's well-defined and documented.

Reply:
@krisgesling I am purposefully trying to avoid the request/response mechanism.  in my reading on event driven architecture, the proper way to respond to a command or request is to emit an event indicating the task was completed.

Reply:
@MetaGamer The current definition is whether or not any alarms exists that have not expired.  I will add more information in the docstring.

Reply:
FWIW, the HiveMind spec, which is a kind of superset of the current Mycroft stack, splits the difference by emitting a response containing `ACK INSTRUCTION`

(doesn't translate directly, because `INSTRUCTION` is an index, but it accomplishes cleanliness while preserving the two-way bus)